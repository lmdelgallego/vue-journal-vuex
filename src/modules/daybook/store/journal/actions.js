/*
export const myAction = async ({commit}) => {

}
*/

import journalApi from '@/api/journalApi';
// import { deleteImage } from '@/helpers/imageApi';

export const loadEntries = async ({commit}) => {
  const {data} = await journalApi.get('/entries.json');

  if(!data) {
    commit('setEntries', []);
    return;
  }

  const entries = [];
  for (let id of Object.keys(data))  {
      entries.push({id, ...data[id]})
  }
  commit('setEntries', entries);
}

export const createEntry = async ({commit}, entry) => {
  const {date, picture, text} = entry;
  const dataToSave = {date,picture,text};
  const {data} = await journalApi.post(`/entries.json`, dataToSave );
  commit('addEntries', {...entry, id: data.name });
  return data.name;
}

export const updateEntry = async ({commit}, entry) => {
  const {date, picture, text} = entry;
  const dataToSave = {date,picture,text};
  await journalApi.put(`/entries/${entry.id}.json`, dataToSave );
  dataToSave.id = entry.id;
  commit('updateEntries', {...dataToSave});
}

//delete entry
export const removeEntry = async ({commit}, entry) => {
  await journalApi.delete(`/entries/${entry.id}.json`);
  // await deleteImage(entry.picture_id);
  commit('removeEntry', entry.id);
}