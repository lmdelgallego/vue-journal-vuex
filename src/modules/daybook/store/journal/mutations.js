/*
export const myMutation = (state) => {

}
* */

export const setEntries = (state, entries) => {
  state.entries = [...state.entries, ...entries];
  state.isLoading = false;
}

export const updateEntries = (/*state*/) => {

}
export const addEntries = (/*state*/) => {

}
