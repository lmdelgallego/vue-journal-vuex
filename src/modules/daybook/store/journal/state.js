export default () => ({
  isLoading: true,
  entries: [
    {
      id: '1',
      date: new Date().toDateString(),
      text:
        'Anim anim consequat ea sit excepteur consectetur minim Lorem amet nostrud. Id ea ut occaecat officia. Velit non anim tempor quis nulla id ea. Exercitation minim esse eu enim ut nisi incididunt Lorem. Minim in duis commodo consectetur sint anim ut consequat. Duis laborum consectetur mollit dolor culpa minim.',
      picture: null,
    },
    {
      id: '2',
      date: new Date().toDateString(),
      text:
        'Magna in enim cupidatat et tempor. Adipisicing duis culpa est mollit veniam id fugiat Lorem aliquip. Laboris deserunt amet occaecat reprehenderit.',
      picture: null,
    },
    {
      id: '3',
      date: new Date().toDateString(),
      text:
        'Esse duis aliqua nisi aliqua do mollit tempor adipisicing occaecat do laboris ad. Nostrud et deserunt consectetur commodo Lorem deserunt ipsum. Cillum magna adipisicing reprehenderit quis adipisicing ad tempor est aliquip quis veniam sint magna ullamco. Duis non elit cupidatat eiusmod magna adipisicing minim consectetur exercitation dolor culpa occaecat esse nisi. Velit Lorem excepteur esse anim nulla incididunt. Qui tempor voluptate esse cillum elit consectetur nisi eu eu dolore.',
      picture: null,
    },
    {
      id: '4',
      date: new Date().toDateString(),
      text:
        'EVeniam dolor enim amet cupidatat cillum officia reprehenderit pariatur. Proident adipisicing mollit culpa eu. Velit tempor enim in nostrud commodo esse eu proident dolor proident qui occaecat officia. Do in ex tempor nulla ex in. Minim Lorem ad commodo elit ea id voluptate ut ut.',
      picture: null,
    },
    {
      id: '5',
      date: new Date().toDateString(),
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicin',
      picture: null,
    },
  ],
});
