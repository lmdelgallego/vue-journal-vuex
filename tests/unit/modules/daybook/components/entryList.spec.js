import { createStore } from "vuex";
import { shallowMount } from "@vue/test-utils";

import EntryList from '@/modules/daybook/components/Entrylist.vue';
import journal from '@/modules/daybook/store/journal';
import {testJournalState} from '../../../mocks/test-journal-state';



const createVuexStore = (initialState) => createStore({
  modules: {
    journal: {
      ...journal,
      state: {...initialState}
    },
  }
});

const mockRoute = {
  push: jest.fn(),
};

/*
const journalMockModule = {
    namespaced: true,
    getters: {
      // getEntriesByTerm: jest.fn()
      getEntryByTerm,
      getEntryById
    },
    state: () =>({
      isLoading: false,
      entries: testJournalState.entries
    })
  };

  const store = createStore({
    modules: {
      journal: {...journalMockModule}
    }
  }); */

describe('EntryList.vue', () => {

  const store = createVuexStore(testJournalState);

  let wrapper ;

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRoute
        },
        plugins: [store],
      }
    });
  });

  test('should be call getEntriesByTerm without terms and show two entries', () => {
    expect(wrapper.findAll('entry-stub').length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should be call getEntryByTerm and filter entries', async () => {
    const input = wrapper.find('input');
    await input.setValue('primer');
    expect(wrapper.findAll('entry-stub').length).toBe(1);
  });

  test('should be redirect to /new when the button is clicked ', () => {
     wrapper.find('button').trigger('click');
     expect(mockRoute.push).toHaveBeenCalled();
     expect(mockRoute.push).toHaveBeenCalledWith({"name": "daybook-entry", "params": {"id": "new"}});
  })

});