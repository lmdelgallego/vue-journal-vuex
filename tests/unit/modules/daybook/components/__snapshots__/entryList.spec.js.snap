// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryList.vue should be call getEntriesByTerm without terms and show two entries 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Buscar"></div>
  <div class="mt-2 d-flex flex-column"><button type="button" class="btn btn-primary mx-3 mb-2"><i class="fa fa-plus-circle"></i> Nueva Entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
